spotbugs/etc/messages.xml

```diff
@@ -1652,6 +1652,90 @@ factory pattern to create these objects.</p>
       ]]>
       </Details>
   </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.ForbiddenSystemOut">
+    <Details>
+      <![CDATA[
+        <p> Can not use System.out and System.err .</p>
+      ]]>
+    </Details>
+  </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.ForbiddenHashtable">
+    <Details>
+      <![CDATA[
+       <p> Can not use Hashtable.</p>
+      ]]>
+    </Details>
+  </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.CheckLogParamNum">
+    <Details>
+      <![CDATA[
+        <p> Use org.slf4j.Logger with wrong placeholder count.</p>
+      ]]>
+    </Details>
+  </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.FindLockOnLocal">
+    <Details>
+      <![CDATA[
+        <p> Locked on a local variable.</p>
+      ]]>
+    </Details>
+  </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.LocalVariableAssignmentCheck">
+    <Details>
+      <![CDATA[ 
+        <p>The assignment of local variables in all branches is the same, which may be incorrect.</p>
+      ]]>  
+    </Details>
+  </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.MethodParamNullCheck">
+    <Details>
+      <![CDATA[
+        <p>The public method should check the parameter not null before using it.</p>
+      ]]>
+    </Details>
+  </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.ChildListNeedsOrderBy">
+    <Details>
+      <![CDATA[
+        <p>Subtables of "List" type may require OrderBy annotation.</p>
+      ]]>
+    </Details>
+  </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.MapIteratorWayCheck">
+    <Details>
+      <![CDATA[
+        <p>The way to iterator map is inefficient.</p>
+      ]]>
+    </Details>
+  </Detector>
+   <Detector class="edu.umd.cs.findbugs.detect.FinallyReturnCheck">
+    <Details>
+      <![CDATA[
+        <p>It is risky to return value in finally block.</p>
+      ]]>
+    </Details>
+  </Detector>
+  <Detector class="edu.umd.cs.findbugs.detect.ObjectNoEqualsAndHashCode">
+    <Details>
+      <![CDATA[
+        <p>The object doesn't rewrriten equals and hashCode methods.</p>
```

Comment:
Misspelled "rewrite".

Note that the `fb-contrib` plugin already has detectors for this.